30413/Chindea Tudor - Eugen/Assgnment_2/Assignment_2/main.c

```diff
@@ -0,0 +1,240 @@
+    oldNode=sentinel->next->next;
+    free(sentinel->next);
+    sentinel->next=oldNode;
+    oldNode->prev=sentinel;
+}
+void DL()
+{
+
+    NodeT * oldNode=sentinel->prev->prev;
+    if(sentinel->prev!=sentinel);
+    {
+    free(sentinel->prev);
+    sentinel->prev=oldNode;
+    oldNode->next=sentinel;
+    }
+}
+void PRINT_ALL(FILE *f)
+{
+
+    if(sentinel->next==sentinel)
+    {
+        fprintf(f,"List is empty\n");
+    }
+    else
+    {
+        NodeT * currentNode= sentinel->next;
+        while(currentNode!=sentinel)
+        {
+            fprintf(f,"%d ",currentNode->data);
+            currentNode=currentNode->next;
+        }
+        fprintf(f,"\n");
+    }
+}
+void PRINT_CONSOLE()
+{
+
+    if(sentinel->next==sentinel)
+    {
+        printf("List is empty\n");
+    }
+    else
+    {
+        NodeT * currentNode= sentinel->next;
+        while(currentNode!=sentinel)
+        {
+            printf("%d ",currentNode->data);
+            currentNode=currentNode->next;
+        }
+        printf("\n");
+    }
+}
+void DOOM_THE_LIST()
+{
+    NodeT *currentNode=sentinel->next;
+    while(currentNode!=sentinel)
+    {
+        sentinel->next=currentNode->next;
+        free(currentNode);
+        currentNode=sentinel->next;
+    }
+   initializeSentinel();
+}
+void DE(int data)
+{
+    NodeT* currentNode=sentinel->next;
+    NodeT * previousNode=sentinel->next;
+    while(currentNode!=sentinel)
+    {
+        if(currentNode->data==data)
+        {
+            if(currentNode == sentinel->next)
+            {
+                sentinel->next=currentNode->next;
+            }
+            else if(currentNode== sentinel->prev)
+            {
+                sentinel->prev=previousNode;
+            }
+            else
+            {
+                previousNode->next = currentNode->next;
+                currentNode->prev=currentNode;
+            }
+            previousNode=currentNode;
+            currentNode=currentNode->next;
+            free(previousNode);
+            previousNode=currentNode;
+        }
+        else
+        {
+            previousNode=currentNode;
+            currentNode=currentNode->next;
+        }
+    }
+}
+void PRIINT_F(int number,FILE *f)
+{
+     int i=0;
+    NodeT * currentNode=sentinel->next;
+    while( currentNode != sentinel && i<number)
+    {
+        fprintf (f,"%d ",currentNode->data);
+        currentNode=currentNode->next;
+        i++;
+    }
+    fprintf(f,"\n");
+}
+void PRINT_L(int nr, FILE *f)
+{
+    int total=0;
+    NodeT * currentNode=sentinel->next;
+    while (currentNode!= sentinel)
+    {
+        total++;
+        currentNode=currentNode->next;
+    }
+    int ct=0;
+    currentNode=sentinel->next;
+    while(currentNode != sentinel)
+    {
+        if(total-ct<=nr)
+        {
+            fprintf(f,"%d ",currentNode->data);
+        }
+        currentNode=currentNode->next;
+        ct++;
+    }
+    fprintf(f,"\n");
+}
+void instructiuni(FILE *f,FILE *g)
```

Comment:
use english for the names you give to variables, methods  and any other piece of code you write